Ship/Ship.Domain/Interfaces/IRepository.cs

```diff
@@ -0,0 +1,12 @@
+using System;
+namespace Ship.Domain.Interfaces;
+
+public interface IRepository<T, K>
+{
+	Task<T?> GetByIdAsync(K id);
+	Task<IEnumerable<T>> GetAllAsync(IEnumerable<ISpecification<T>> specs);
+	Task<T> CreateAsync(T entity);
```

Comment:
when use async methods pass cancellation token starting from controller to service and then repository